src/ring_theory/tensor_product.lean

```diff
@@ -408,11 +408,9 @@ instance left_algebra : algebra S (A ⊗[R] B) :=
 instance : algebra R (A ⊗[R] B) := infer_instance
 
 @[simp]
-lemma algebra_map_apply (r : S) :
+lemma algebra_map_apply [smul_comm_class R S A] (r : S) :
   (algebra_map S (A ⊗[R] B)) r = ((algebra_map S A) r) ⊗ₜ 1 := rfl
 
-instance : is_scalar_tower R S (A ⊗[R] B) := ⟨λ a b c, by simp⟩
```

Comment:
Where did this instance go? Is it true by `apply_instance`? If so, please mention that in the PR description!

Reply:
Yes, both of the following work:
```lean
instance [algebra R S] [is_scalar_tower R S A] : is_scalar_tower R S (A ⊗[R] B) := infer_instance
instance [algebra S R] [is_scalar_tower S R A] : is_scalar_tower S R (A ⊗[R] B) := infer_instance
```
I added it to the PR description.

Reply:
Since this creates a forward-porting obligation anyway, should we add the missing results:

* `is_scalar_tower S (A ⊗[R] B) (A ⊗[R] B)`
* `smul_comm_class S (A ⊗[R] B) (A ⊗[R] B)`

for when `A` is a non-unital algebra?

Reply:
These are the two instances I was thinking of:

```lean
instance [monoid S] [distrib_mul_action S A] [is_scalar_tower S A A] [smul_comm_class R S A] :
  is_scalar_tower S (A ⊗[R] B) (A ⊗[R] B) :=
{ smul_assoc := λ r x y, begin
    change (r • x) * y = r • (x * y),
    apply tensor_product.induction_on y,
    { simp [smul_zero], },
    { apply tensor_product.induction_on x,
      { simp [smul_zero] },
      { intros a b a' b',
        dsimp,
        rw [tensor_product.smul_tmul', tensor_product.smul_tmul', tmul_mul_tmul, smul_mul_assoc], },
      { intros, simp [smul_add, add_mul, *], } },
    { intros, simp [smul_add, mul_add, *], },
  end }

instance [monoid S] [distrib_mul_action S A] [smul_comm_class S A A] [smul_comm_class R S A] :
  smul_comm_class S (A ⊗[R] B) (A ⊗[R] B) :=
{ smul_comm := λ r x y, begin
    change r • (x * y) = x * r • y,
    apply tensor_product.induction_on y,
    { simp [smul_zero], },
    { apply tensor_product.induction_on x,
      { simp [smul_zero] },
      { intros a b a' b',
        dsimp,
        rw [tensor_product.smul_tmul', tensor_product.smul_tmul', tmul_mul_tmul, mul_smul_comm], },
      { intros, simp [smul_add, add_mul, *], } },
    { intros, simp [smul_add, mul_add, *], },
  end }
  ```

Reply:
(#19143)